sdk/iot/core/src/az_iot_hub_client.c

```diff
@@ -107,3 +108,40 @@ AZ_NODISCARD az_result az_iot_hub_client_id_get(
 
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result
+az_iot_hub_client_properties_init(az_iot_hub_client_properties* properties, az_span buffer)
+{
+  AZ_PRECONDITION_NOT_NULL(properties);
+  AZ_PRECONDITION_VALID_SPAN(buffer, 0, false);
+
+  properties->_internal.properties = buffer;
+  properties->_internal.current_property = az_span_ptr(buffer);
+
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result az_iot_hub_client_properties_append(
+    az_iot_hub_client_properties* properties,
+    az_span name,
+    az_span value)
+{
```

Comment:
how about exposing system properties keynames from core?

Reply:
As in which are available? Whats the context here

Reply:
like content type, 
correlation-id
content-encoding
message-id

Reply:
Yea we definitely can have them as needed. I'd like to add them on an as needed basis as we roll features out.